From 52fdc9393c716ff9ed4ae0ffac17eaf5d01ec50b Mon Sep 17 00:00:00 2001
From: Mike Kasick <mike@kasick.org>
Date: Sun, 1 Mar 2026 16:20:15 -0500
Subject: [PATCH] retroarch: command_event: CMD_EVENT_FULLSCREEN_TOGGLE: Toggle
 video_scale_integer.

---
 retroarch.c | 61 +++--------------------------------------------------
 1 file changed, 3 insertions(+), 58 deletions(-)

diff --git a/retroarch.c b/retroarch.c
index b69e1ab2ad..ceedd86b38 100644
--- a/retroarch.c
+++ b/retroarch.c
@@ -5040,64 +5040,9 @@ bool command_event(enum event_command cmd, void *data)
          return false;
 #endif
       case CMD_EVENT_FULLSCREEN_TOGGLE:
-         {
-            audio_driver_state_t
-               *audio_st              = audio_state_get_ptr();
-            input_driver_state_t
-               *input_st              = input_state_get_ptr();
-            bool *userdata            = (bool*)data;
-            bool video_fullscreen     = settings->bools.video_fullscreen;
-            bool ra_is_forced_fs      = (video_st->flags &
-               VIDEO_FLAG_FORCE_FULLSCREEN) ? true : false;
-            bool new_fullscreen_state = !video_fullscreen && !ra_is_forced_fs;
-
-            if (!video_driver_has_windowed())
-               return false;
-
-            audio_st->flags |= AUDIO_FLAG_SUSPENDED;
-            video_st->flags |= VIDEO_FLAG_IS_SWITCHING_DISPLAY_MODE;
-
-            /* we toggled manually, write the new value to settings */
-            configuration_set_bool(settings, settings->bools.video_fullscreen,
-                  new_fullscreen_state);
-            /* Need to grab this setting's value again */
-            video_fullscreen = new_fullscreen_state;
-
-            /* we toggled manually, the CLI arg is irrelevant now */
-            if (ra_is_forced_fs)
-               video_st->flags &= ~VIDEO_FLAG_FORCE_FULLSCREEN;
-
-            /* If we go fullscreen we drop all drivers and
-             * reinitialize to be safe. */
-            command_event(CMD_EVENT_REINIT, NULL);
-            if (video_fullscreen)
-            {
-               if (     video_st->poke
-                     && video_st->poke->show_mouse)
-                  video_st->poke->show_mouse(video_st->data, false);
-               if (!settings->bools.video_windowed_fullscreen)
-                  if (input_driver_grab_mouse())
-                     input_st->flags |= INP_FLAG_GRAB_MOUSE_STATE;
-            }
-            else
-            {
-               if (     video_st->poke
-                     && video_st->poke->show_mouse)
-                  video_st->poke->show_mouse(video_st->data, true);
-               if (!settings->bools.video_windowed_fullscreen)
-                  if (input_driver_ungrab_mouse())
-                     input_st->flags &= ~INP_FLAG_GRAB_MOUSE_STATE;
-            }
-#ifdef HAVE_OVERLAY
-            input_overlay_check_mouse_cursor();
-#endif
-
-            video_st->flags &= ~VIDEO_FLAG_IS_SWITCHING_DISPLAY_MODE;
-            audio_st->flags &= ~AUDIO_FLAG_SUSPENDED;
-
-            if (userdata && *userdata == true)
-               video_driver_cached_frame();
-         }
+         /* Repurpose for toggling integer scale. */
+         settings->bools.video_scale_integer = !settings->bools.video_scale_integer;
+         command_event(CMD_EVENT_VIDEO_APPLY_STATE_CHANGES, NULL);
          break;
       case CMD_EVENT_DISK_APPEND_IMAGE:
          {
-- 
2.43.0

